### ISSUE TITLE
Challenge restart timing inconsistency

### ISSUE THREAD

**Issue by ChallengeSY** [Neverball/neverball#329]
An inconsistency was discovered with respect to Challenge mode runs.

Time outs and fall outs do correctly count against the set time, but restarts of successful attempts do _not_; making them possible to abuse, especially on easier sets.

**Comment by parasti** [comment]
This is by design. If Neverforum was still around, I could probably find the link where this was discussed. I don't really understand what you mean by abuse. You can improve your whole-set run, but that's not really abuse IMO.

**Comment by ChallengeSY** [comment]
> You can improve your whole-set run, but that's not really abuse IMO.
That is true, but improving a set run as a whole means starting fresh from the beginning.

Rather, it is restarting a single level to improve its time which "incorrectly" reduces the time of the whole set. For example, if I finish a "Level 10" with 0'15''99, but then improve it to 0'14''49 in the same run, the final set time factors in **only** the 0'14''49

**Comment by parasti** [comment]
Yes, what you're describing is the intended behavior. Only the last playthrough of the level is included in the set time.

If you don't want that, you can set a personal, arbitrary rule of never restarting previous levels.

Maybe I'm missing something but I believe this is working as intended.

**Comment by ChallengeSY** [comment]
There was a discussion in the updated Nevercord's _neverball-talk_ on Feb 25; resulted in a Medium **speedrun** being rejected because it included a "voluntary restart".

Ideally, such a run should one-shot each of the regular levels anyway; the rejected run had 4 fall outs throughout the set anyway.

**Comment by parasti** [comment]
Well, if a speedrun competition sets a rule that you can't restart completed levels, that warrants a disqualification.

On the other hand, restarting the level and getting a better result (**without** a fall-out or time-out in between) seems like a worthy achievement to me.

Do you get infinite restarts even if you fall out after the first restart?

**Comment by ChallengeSY** [comment]
> Do you get infinite restarts even if you fall out after the first restart?

Nope, regardless of the outcome of the first try, a fall out (or a time out) on any subsequent attempt will still deduct a ball from the available stock. This makes the harder sets, well, harder to "abuse".

Anyhow, as far as this issue is concerned, I can't think of anything else. I _might_ play around with the code sometime; just not at present.
